relentless/optimize/objective.py

```diff
@@ -165,3 +168,66 @@ def design_variables(self):
         :class:`ObjectiveFunction` for which the result was constructed, mapped
         to the value of the variables at the time the result was constructed."""
         return self._design_variables
+
+class RelativeEntropy(ObjectiveFunction):
+    def __init__(self, target, simulation, potentials, thermo, dr, communicator=None):
+        if target.V is None or target.N is None:
+            raise ValueError('The target ensemble must have both V and N set.')
+
+        self.target = target
+        self.simulation = simulation
+        self.potentials = potentials
+        self.thermo = thermo
+        self.dr = dr
+        self.communicator = communicator
+
+    def compute(self, directory=None):
+        sim = self.simulation.run(self.target, self.potentials, directory, self.communicator)
+        sim_ens = self.thermo.extract_ensemble(sim)
+
+        # compute the relative entropy gradient by integration
+        g_tgt = self.target.rdf
+        g_sim = sim_ens.rdf
+        dvars = self.design_variables()
+        gradient = _collections.KeyedArray(keys=dvars)
+
+        for var in dvars:
+            update = 0
+            for i,j in self.target.pairs:
+                # evaluate derivative wrt design variable and interpolate with r
+                rs = self.potentials.pair.r
+                dus = self.potentials.pair.derivative((i,j),var)
+                for k,duk in enumerate(dus):
+                    if np.isinf(duk):
+                        rs = np.delete(rs,k)
+                        dus = np.delete(dus,k)
+                        break
```

Comment:
This is the right idea, but I think you can do this faster:
```py
flags = np.isinf(u)
first_finite = 0
while not flags[first_finite] and first_finite < len(rs):
    first_finite += 1
rs = rs[first_finite:]
dus = dus[first_finite:]
```
(There might be a way to do this without a loop too, but at least here you are not copy the array repeatedly)